testsuite/cluster-test/src/experiments/load_test.rs

```diff
@@ -0,0 +1,327 @@
+impl Experiment for LoadTest {
+    fn affected_validators(&self) -> HashSet<String> {
+        HashSet::new()
+    }
+    async fn run(&mut self, context: &mut Context<'_>) -> anyhow::Result<()> {
+        // spin up StubbedNode
+        let vfn = context.cluster.random_fullnode_instance();
+        let vfn_endpoint = format!("http://{}:{}/v1", vfn.ip(), vfn.ac_port());
+
+        let mut stubbed_node = StubbedNode::launch(vfn_endpoint).await;
+
+        let mut emit_job = None;
+        let mut mempool_task = None;
+        let mut state_sync_task = None;
+        let duration = Duration::from_secs(self.duration);
+
+        if self.emit_txn {
+            // emit txns to JSON RPC
+            // spawn future
+            emit_job = Some(
+                context
+                    .tx_emitter
+                    .start_job(EmitJobRequest::for_instances(
+                        context.cluster.fullnode_instances().to_vec(),
+                        context.global_emit_job_request,
+                    ))
+                    .await?,
+            );
+        }
+
+        if self.mempool {
+            // spawn mempool load test
+            let (mempool_sender, mempool_events) = stubbed_node
+                .mempool_handle
+                .take()
+                .expect("missing mempool network handles");
+            mempool_task = Some(tokio::task::spawn(mempool_load_test(
+                duration,
+                mempool_sender,
+                mempool_events,
+            )));
+        }
+
+        if self.state_sync {
+            // spawn state sync load test
+            let (state_sync_sender, state_sync_events) = stubbed_node
+                .state_sync_handle
+                .take()
+                .expect("missing state sync network handles");
+            state_sync_task = Some(tokio::task::spawn(state_sync_load_test(
+                duration,
+                state_sync_sender,
+                state_sync_events,
+            )));
+        }
+
+        // await on all spawned tasks
+        tokio::time::delay_for(Duration::from_secs(self.duration)).await;
+        if let Some(j) = emit_job {
+            let _ = context.tx_emitter.stop_job(j).await;
+        }
+
+        if let Some(t) = mempool_task {
+            let _ = t.await.expect("failed mempool load test task");
+        }
+
+        if let Some(t) = state_sync_task {
+            let _ = t.await.expect("failed state sync load test task");
+        }
+
+        // TODO log per-component experiment results
+
+        Ok(())
+    }
+    fn deadline(&self) -> Duration {
+        Duration::from_secs(self.duration)
+    }
+}
+
+impl fmt::Display for LoadTest {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "Load test components: mempool: {}, state sync: {}, emit txns: {}",
+            self.mempool, self.state_sync, self.emit_txn,
+        )
+    }
+}
+
+// An actor that can participate in LibraNet
```

Comment:
nit: I think we can move this struct and impl to a separate file under /cluster-test/ or any reasonable place